### DISCUSSION TITLE
How to make a Text to Item that works at island?

### DISCUSSION THREAD

**Discussion by SirEnder125** [RedShyGuy/Vapecord-ACNL-Plugin#30] [Q&A] [answered]
Hello! I recently made a few cheats and I tried to make text to item, coord mod, etc.
I ran into a few problems. That is, they work fine at my town, but not at all online.
I would guess that I would need a pointer to the player instance by which I could
get the current inventory slot address... How would I go about doing that?

**Comment by RedShyGuy** [comment, marked as answer]
Hey!
Yes you do need a pointer which always points to the right Player Save Data.
The easiest way to find that pointer is by searching for the inventory save data on the island and on the town. Then calculate the difference of them. And then do a new Search. And after switching the location (town to island) search for the difference. Only the Player Save Data Pointer should appear then.

**Reply by SirEnder125** [reply]
~Hey, so I found the town inventory address, and went to the offline island.
The difference was 8... Is this right? Should I do the online one instead?~

EDIT 1: Hey, I got two addresses, 0x31F2DBF0 and 0xA8C360. I subtracted the second
from the first, did a CTRPF search for the difference, with not hits. Why?

EDIT 2: I think I got it wrong again... I got 4 addresses:
0x307815D4,
0x30A1199C,
0x30A92FD4, and
0x30ACFAD4

**Reply by RedShyGuy** [reply]
```cpp
u32 GetPlayerSaveData(void) {
    return *(u32 *)0xAB014C;
}
```
Well here, this will return the Player Save Address. You will need to jump the the inventory though. The Inventory is at `0x6BD0` in the player data, so:
```cpp
u32 Inv = GetPlayerSaveData() + 0x6BD0;
```
Will give you the inventory address

**Reply by SirEnder125** [reply]
Oh, okay. Thanks. Though, I do like to find the addresses myself, just so that I am aware of what
I am doing, where I'm editing. Thank you, though. I'll use this. :)

I don't want to feel that I am too dumb, though. So let me see if I have this straight:
1. Search for Island inventory address and town inventory address
2. Find the difference between them.
3. Make a new search with the value set to the difference found in step 2?

Somehow that sounds wrong... Is it?

**Reply by SirEnder125** [reply]
Nevermind, I figured it out! :D

EDIT: Nevermind, I am still having trouble.
I know exactly what to do. But, every time I do it, it is about to show the one search hit...
and it crashes. Or sometimes freezes... Why??

**Reply by RedShyGuy** [reply]
the search is a very complex thing, it can sometimes crash yea

**Reply by SirEnder125** [reply]
But now it is crashing every time I search for anything. It crashes on both of my
2ds's. On one game only as far as I know. And on every CTRPF plugin I load into it.
I've tried searching for ANYTHING and it ALWAYS crashes.

**Reply by RedShyGuy** [reply]
Thats weird, try deleting the search folder in your plugin directory

**Reply by SirEnder125** [reply]
Okay, I'll try that.

**Reply by SirEnder125** [reply]
It did not change a thing. :/

**Reply by RedShyGuy** [reply]
Then I am really not sure sorry :/

**Reply by SirEnder125** [reply]
Oh, well. Thanks for trying. :wink:

**Reply by SirEnder125** [reply]
Not sure if this helps but here is footage of my game crashing during search for the address.
[Video](https://streamable.com/jqdkys)

Although there was someone at that specific island being very annoying
and I'm kind of glad it crashed, it does make things a bit difficult.

**Reply by RedShyGuy** [reply]
I see, this exact issue did happen to me lots of times already. But I have to admit I have no idea why it happens and how to make it not happen anymore. Sometimes it works for me and rarely it just doesnt at all.

**Reply by SirEnder125** [reply]
Oh, okay. I'll keep searching for the cause, and tell you if I find any way to avoid these problems. :P
